lms/resources/_js_config.py

```diff
@@ -24,6 +24,7 @@ def __init__(self, context, request):
         self._grading_info_service = request.find_service(name="grading_info")
         self._ai_getter = request.find_service(name="ai_getter")
         self._h_api = request.find_service(name="h_api")
+        self._lti_h = request.find_service(name="lti_h")
```

Comment:
This seems a bit squiffy. This class doesn't want the LTI-H service, just an `HUser`. I think we're using the `LTI-H` service here because it's gained a `h_user` property, which just calls `from_lti_user`,but we only need _that_ because the `LTIUser` doesn't know the authority.

I think we can simplify this as per our discussion:

 * Let the `LTIUser` conceptually be all user data in an LTI context (rather than a user explictly created from or representing LTI params). We started this by adding the `display_name`
 * Add authority to the `LTIUser`
 * Move `h_user` to be a property of `LTIUser`

The result in this class would be `self._lti_user.h_user`

Reply:
Yes, I've expanded the responsibility of `LTIHService` to include generating `HUser`'s from `LTIUser`'s, which seems an appropriate responsibility for the LTI -> H translation service. It's just `HUser.from_lti_user(request.lti_user, request.registry.settings["h_authority"])` though, so this property could be put anywhere that has access to the `request`. For example `request.h_user` should be doable.

I think adding `authority` to `LTIUser` is controversial in a couple of ways:

1. `authority` is an h concept, not an LTI concept, and `LTIUser` is `LTIUser` not `HUser`. It's specifically about the LTI user in what it represents and how it's used, currently.

   When an `HUser` is generated from an `LTIUser` that seems like the appropriate time for `authority` to come in to the picture, because LTI users don't have authorities but h users do. So `HUser.from_lti_user(lti_user, authority)` seems appropriate to me. Those are the correct two ingredients: everything needed for an LMS app h user comes from two ingredients, an LTI user and the LMS app's authority setting. Conceptually it's actually _not_ the case that you can generate an h user from just an LTI user.

   Turning `LTIUser` into just `User` seems like a much bigger move to me, and I'm not sure about it. `LTIUser` is specifically about LTI users in what it represents and what it's used for. I think it would need more thinking through. It certainly goes beyond what I need to do right now (which is to decouple `LTIHService` from `LTILaunchResource` so that the sync API can call it).

2. Even though it has crept into `models.HUser` already, I think `authority` should likely be treated as a deployment setting, not as part of the domain model. There's only ever one authority in play, which is `request.registry.settings["h_authority"]` (and is always `"lms.hypothes.is"`). There aren't different users and groups with different authorities. It's a setting.

   This might be the correct way to enable an `LTIUser.h_user` property to exist: remove the `authority` attribute from `HUser` so it's gone from the domain model entirely. Just have services that actually need to pass the authority to h (`HAPIService`) read the authority setting. But that's a bigger refactoring that I don't want to bite off yet

Reply:
If we can get rid of authority from both, then I think that's the best outcome. I recommended renaming "LTIUser" to get around this issue. It would become what we need it to be to get the job done, rather than being specific to LTI. But as you say, it's probably more akin to a global variable, rather than copying it into every group and user.

I'm less happy about expanding `LTIH`. It was originally called "LTIHSyncService" or something more long winded, and is/was specifically about ferrying data from LMS to H for it to store. A lot of the time that's not what the `HUser` is used for, so I think it sticks out like a sore thumb. `LTIH `currently has a pretty tightly defined role, and I don't think it's made better here. It feels like this needed a home, and this was the closest.

This is probably a small, but distinct service. As per our discussion in H, as this only returns one thing, we would follow the service factory pattern where you just get an `HUser` right away:

```self._h_user = request.find_service(name="h_user")```

If we eliminate the factory method on the user object it won't seem quite as pathetic.